_posts/2019-02-05-configurable-builds-part-1.md

```diff
@@ -0,0 +1,280 @@
+---
+layout: posts
+title: Configurable Builds - Part 1
+authors:
+ - gregestren
+---
+
+The larger your project gets, the more likely you have to build it in different
+ways.
+
+Maybe you need Android, iOS, and desktop versions of your app. Maybe you build
+C++ for different platforms. Or maybe you maintain a popular library that
+different users need different features from.
```

Comment:
How's this? It's not my ideal example, since JavaScript isn't the most obvious use case for a build tool. But it's the only tech I can think of that makes its point concisely.

The broader scenario I'm trying to express is "feature flags", which is a concept we already have demand for and support. It's a different use case than the other examples which illustrate platforms, and my attempt to illustrate that configurability goes beyond just platforms.  

"Features" are project or domain-specific: maybe "use native extensions vs. pure bytecode" or "include experimental app features" or "enterprise vs. free version", etc. Libraries advertise a bunch of these "knobs" that binaries can toggle according to their specific preferences.

Reply:
I like it, it's much more concrete and understandable. I agree with the previous comment that it's still a little hard to read. What about: "Or maybe you maintain a JavaScript library, and your users only want to build specific modules from it"?

Reply:
Okay, I integrated the essence of your suggestion.

I kept *popular* because, as I read these themes, that's the distinction between "I maintain a js_library for my js_binary" vs. "I maintain a professionally maintained library with a clear API aimed  at lots of disparate users". Others may read these themes differently but it reads more confusing to me without that.

I also tried to shorten the phrasing, keep the sentence single-clause, and end it as concretely as possible (admittedly "from" isn't a great ending word, but I'm even more scared of "it"!).

Reply:
Looks great to me, thank you.